app/routes/todos.tsx

```diff
@@ -44,94 +58,201 @@ export function meta() {
+    }
+  };
+}
+
+function KanbanCard({ todo }: { todo: Todo }) {
+  const { setDraggedId } = useKanban();
+  const formBridge = FormBridge();
+
+  const handleDragStart = (e: React.DragEvent) => {
+    setDraggedId(todo.id);
+    e.dataTransfer.setData('text/plain', todo.id);
+  };
+
+  const handleDragEnd = () => {
+    setDraggedId(null);
+  };
+
+  return (
+    <div
+      draggable
+      onDragStart={handleDragStart}
+      onDragEnd={handleDragEnd}
+      className="p-3 border border-gray-200 rounded-md bg-white shadow-sm mb-2 cursor-move"
+      style={{ minHeight: '80px' }}
+      data-todo-id={todo.id}
+    >
+      <div className="flex items-start justify-between">
+        <span className="text-gray-800 text-sm flex-1">{todo.text}</span>
+        <div className="flex gap-1 ml-2">
+          <button
+            onClick={() => formBridge.submit({ intent: 'move', id: todo.id, target: 'done' })}
+            className="text-xs px-2 py-1 bg-blue-100 text-blue-600 rounded hover:bg-blue-200"
+          >
+            Move
+          </button>
+          <Form method="post" style={{ display: "contents" }}>
+            <input type="hidden" name="intent" value="delete" />
+            <input type="hidden" name="id" value={todo.id} />
             <button
               type="submit"
-              className="px-4 py-2 bg-blue-500 text-white rounded-md hover:bg-blue-600 focus:outline-none focus:ring-2 focus:ring-blue-500"
+              className="text-red-500 hover:text-red-700 font-bold text-sm"
             >
-              Add
+              ×
             </button>
+          </Form>
+        </div>
+      </div>
+    </div>
+  );
+}
+
+function KanbanColumn({ 
+  status, 
+  todos, 
+  title 
+}: { 
+  status: KanbanStatus; 
+  todos: Todo[]; 
+  title: string; 
+}) {
+  const formBridge = FormBridge();
+
+  const handleDragOver = (e: React.DragEvent) => {
+    e.preventDefault();
+  };
+
+  const handleDrop = (e: React.DragEvent) => {
+    e.preventDefault();
+    const id = e.dataTransfer.getData('text/plain');
+    formBridge.submit({ intent: 'move', id, target: status });
+  };
+
+  return (
+    <div
+      className="flex-1 p-4 bg-gray-100 rounded-lg"
+      style={{ minHeight: '400px' }}
+      onDragOver={handleDragOver}
+      onDrop={handleDrop}
+    >
+      <h3 className="font-semibold text-gray-700 mb-3 text-center">{title}</h3>
+      <div className="space-y-2">
+        {todos.map((todo) => (
+          <KanbanCard key={todo.id} todo={todo} />
+        ))}
+      </div>
+    </div>
+  );
+}
+
+function KanbanBoard({ todos }: { todos: Todo[] }) {
+  const todosByStatus = useMemo(() => {
+    const todoList = todos.filter(t => !t.status || t.status === 'todo' || (!t.completed && !t.status));
+    const doingList = todos.filter(t => t.status === 'doing');
+    const doneList = todos.filter(t => t.status === 'done' || t.completed);
+    
+    return { todo: todoList, doing: doingList, done: doneList };
```

Comment:
issue (complexity): Mixing status and completed will lead to sync issues, for example when todo has status: 'doing' or 'todo'  but completed: true. It can be avoided by filtering directly.
```suggestion
  return {
    todo: todos.filter(t => t.status === 'todo'),
    doing: todos.filter(t => t.status === 'doing'),
    done: todos.filter(t => t.status === 'done'),
  };
```